static-instrumenter/maven-plugin/src/main/java/io/opentelemetry/contrib/staticinstrumenter/plugin/maven/InstrumentationAgent.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.plugin.maven;
+
+import static io.opentelemetry.contrib.staticinstrumenter.plugin.maven.JarSupport.consumeEntries;
+import static io.opentelemetry.contrib.staticinstrumenter.plugin.maven.ZipEntryCreator.moveEntryUpdating;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InstrumentationAgent {
+
+  // TODO: change after the new static-instr capable agent is released
+  public static final String JAR_FILE_NAME = "opentelemetry-javaagent.jar";
```

Comment:
is storing `opentelemetry-javaagent.jar` under source control needed because it's a SNAPSHOT build of the javaagent?

Reply:
It's only temporary - in the final version the agent-with-instrumentation-capabilities (currently - agent with agent-extension) will be fetched during build.